File: Binary Tree/python/depthFirstSearch.py
from node import Node
class DepthFirstSearch:
    def __init__(self, root:Node) -> None:
        self.__root = root
    
    def __inOrder(self, root:Node) ->None:
        if(root ==None):
            return
        
        self.__inOrder(root.left)
        print(root.value, end=",")
        self.__inOrder(root.right)
        
    def displayInOrder(self) -> None:
        print("[", end="")
        self.__inOrder(self.__root)
        print("]")
        
    def __preOredr(self,root:Node) ->None:
        if(root == None):
            return
        
        print(root.value, end=",")
        self.__preOredr(root.left)
        self.__preOredr(root.right)
        
    def displayPreOrder(self) -> None:
        print("[", end="")
        self.__preOredr(self.__root)
        print("]")
        
    def __postOrder(self, root:Node) ->None:
        if(root == None):
            None
            
        self.__postOrder(root.left)
        self.__postOrder(root.right)
        print(root.value, ",")
    
    def displayPostOrder(self) -> None:
        print("[", end="")
        self.__postOrder(self.__root)
        print("]")
        
        
        
    

File: Binary Tree/java/src/DepthFirstSearch.java
public class DepthFirstSearch<T> {
    Node<T> tree;

    DepthFirstSearch(Node<T> tree) {
        this.tree = tree;
    }

    private void inOrder(Node<T> tree) {
        if (tree == null)
            return;
        inOrder(tree.left);
        System.out.print(tree.value + ",");
        inOrder(tree.right);
    }

    void displayInOrder() {
        System.out.print("[");
        inOrder(tree);
        System.out.println("]");
    }

    private void preOrder(Node<T> tree) {
        if (tree == null)
            return;
        System.out.print(tree.value + ",");
        preOrder(tree.left);
        preOrder(tree.right);
    }

    void displayPreOrder() {
        System.out.print("[");
        preOrder(tree);
        System.out.println("]");
    }

    private void postOrder(Node<T> tree) {
        if (tree == null)
            return;
        postOrder(tree.left);
        postOrder(tree.right);
        System.out.print(tree.value + ",");
    }

    void displayPostOrder() {
        System.out.print("[");
        postOrder(tree);
        System.out.println("]");
    }

}
